File: typescript/src/models/volumePricing.ts
// this file is @generated
import { type TierRow, TierRowSerializer } from "./tierRow";

export interface VolumePricing {
  blockSize?: number | null;

  tiers: TierRow[];
}

export const VolumePricingSerializer = {
  _fromJsonObject(object: any): VolumePricing {
    return {
      blockSize: object["block_size"],
      tiers: object["tiers"].map((item: any) => TierRowSerializer._fromJsonObject(item)),
    };
  },

  _toJsonObject(self: VolumePricing): any {
    return {
      block_size: self.blockSize,
      tiers: self.tiers.map((item: any) => TierRowSerializer._toJsonObject(item)),
    };
  },
};

File: python/meteroid/models/volume_pricing.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel
from .tier_row import TierRow


@dataclasses.dataclass
class VolumePricing(BaseModel):
    tiers: t.List[TierRow]

    block_size: t.Optional[int] = None
